### PR TITLE
ci: Update golangci-lint to v1.55.2

### PR DESCRIPTION
**Pull request by nickajacks1**
## Description

Update golangci-lint to the latest version.
New linters and failing rules were commented out for visibility and tagged with TODO comments.

Related to #2816

## Type of Change

Please delete options that are not relevant.

- [x] Code consistency (non-breaking change which improves code reliability and robustness)

## Checklist

Before you submit your pull request, please make sure you meet these requirements:

- [x] Conducted a self-review of the code and provided comments for complex or critical parts.
- [x] Verified that any new dependencies are essential and have been agreed upon by the maintainers/community.

### CODE REVIEW

**Comment by nickajacks1** [comment]
In addition to the new linters mentioned in #2816, several rules and existing linters had a large number of new issues, so I just disabled them for now.

Disabled rules and linters:
- gosec: G104 is now raised for _ assignments (at least when audit: true is set).  Exclude G104 until unchecked errors are ignored (there are a lot).
- depguard: the config format has changed
- revive
  - comment-spacings:  spaces after the // in a comment - there are a few instances of this, including "decorated" comments
  - exported,disableStutterCheck:  some functions, eg csrf.CsrfFromHeader are advised to be changed to csrf.FromHeader. Updating these would be breaking changes, so they should be fixed before v3 is released, and they should be noted in any migration guides.
  - unchecked-type-assertion:  overlaps with errcheck
- goconst: this seems to be more strict now.  A few of the new rules might be a little awkward to fix. we'll see.
- wastedassign: this seems to be more strict now.

Maybe we can drop errcheck in favor of revive's unchecked-type-assertion and gosec's G104? Or the other way around?

**Comment by ReneWerner87** [comment]
> Maybe we can drop errcheck in favor of revive's unchecked-type-assertion and gosec's G104? Or the other way around?

as long as a check is on, that would be okay, the important thing is that the error returns are always checked

File: .golangci.yml [hunk 1]
```diff
@@ -93,13 +95,17 @@ linters-settings:
         disabled: true
       - name: cognitive-complexity
         disabled: true
+      - name: comment-spacings
+        disabled: true # TODO https://github.com/gofiber/fiber/issues/2816
```
Comment (gaby) [inline .golangci.yml:99]: Is this still TODO? Or are we handling this in a diff PR?

Comment (gaby) [reply .golangci.yml:99]: Nevermind :-)

**Review by gaby — APPROVED** [review]
LGTM

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**